language-plutus-core/src/Language/PlutusCore/TypeSynthesis.hs

```diff
@@ -241,7 +241,7 @@ typeOf :: Term TyNameWithKind NameWithType a -> TypeCheckM a (NormalizedType TyN
 typeOf (Var _ (NameWithType (Name (_, ty) _ _))) =
     -- Since we kind check types at lambdas, we can normalize types here without kind checking.
     -- Type normalization at each variable is inefficient and we may consider something else later.
-    normalizeTypeOpt $ void ty
+    rename ty >>= normalizeTypeOpt . void
```

Comment:
Can you add a comment here - it's non-obvious to a reader why you need to do this given that we've renamed the program (as it was non-obvious to us in the past!).

Reply:
Yeah, I clearly should have included a comment, thanks.

Reply:
Done. I plan to [at least try to] remove that annotation stuff, though, as it seems to be the reason why we can't type check dynamic built-ins, so I hope that comment won't be there for too long.